src/modules/royalty.ts

```diff
@@ -0,0 +1,152 @@
+import { Royalty, Royalty__factory } from "@3rdweb/contracts";
+import { BigNumberish } from "ethers";
+import { ModuleType, Role, RolesMap } from "../common";
+import { Currency, getCurrencyMetadata } from "../common/currency";
+import { Module } from "../core/module";
+
+interface IRoyalty {
+  /**
+   * Get the total shares held by payees
+   */
+  totalShares(): Promise<BigNumberish>;
+
+  /**
+   * Get the total amount of royalty already released
+   */
+  totalReleased(): Promise<BigNumberish>;
+
+  /**
+   * Returns the amount of `token` released to payees
+   *
+   * @param tokenAddress - The ERC20 token address to check royalty distributions for
+   */
+  totalReleasedByToken(tokenAddress: string): Promise<BigNumberish>;
+
+  /**
+   * Gets the amount of shares held by an account at the address `address`
+   *
+   * @param address - The address to check the shares held for
+   */
+  shares(address: string): Promise<BigNumberish>;
+
+  /**
+   * Gets the amount of royalties already released to the address `address`
+   *
+   * @param address - The address to check the amount of released royalties
+   */
+  released(address: string): Promise<BigNumberish>;
+
+  /**
+   * Gets the amount of royalties already released to the address `address`
+   * for the specific token `tokenAddress`
+   *
+   * @param address - The address to check the amount of released royalties
+   * @param tokenAddress - The ERC20 token address to check royalty distributions for
+   */
+  releasedByToken(address: string, tokenAddress: string): Promise<BigNumberish>;
+
+  /**
+   * Gets the address of the payee at index `index`
+   *
+   * @param index - The index of the payee to get the address of
+   */
+  payee(index: BigNumberish): Promise<string>;
```

Comment:
i think we probably want something like `getAll` that returns a list of people involves in this maybe including their respective shares.

Reply:
not sure if `payee` is the right name for this too.